### DISCUSSION TITLE
ValidationError not triggered

### DISCUSSION THREAD

**Discussion by raj21082024** [pydantic/pydantic#10869] [Question]
Model:
`from pydantic import BaseModel, Field, ValidationError, root_validator, validator
from typing import List, Optional
import re

class LocationModel(BaseModel):
    provider: str
    id: str
    label: str

    class Config:
        extra = 'forbid'

class AccountIdModel(BaseModel):
    provider: str
    id: str
    label: str

    class Config:
        extra = "forbid"

class TagModel(BaseModel):
    tagKey: str
    tagValue: str

    class Config:
        extra = "forbid"

    @root_validator(pre=True)
    def validate_tag_format(cls, values):
        tag_key = values.get("tagKey")
        tag_value = values.get("tagValue")
        if len(tag_key) > 128:
            raise ValueError("Tag key cannot exceed 128 characters")
        if len(tag_value) > 256:
            raise ValueError("Tag value cannot exceed 256 characters")
        if not re.match(r"^[a-zA-Z0-9_\-\.]+$", tag_key):
            raise ValueError(
                "Tag key can only contain alphanumeric characters, underscores, hyphens, and dots"
            )
        return values

class EventRecipientModel(BaseModel):
    mode: str
    tenant_id: str
    address: str
    name: str
    recipient_id: str

    class Config:
        extra = "forbid"

class EventModel(BaseModel):
    event_recipients: List[EventRecipientModel]

    class Config:
        extra = "forbid"

class ActionModel(BaseModel):
    name: str
    label: str

    class Config:
        extra = "forbid"

class SettingsModel(BaseModel):
    recommendationType: Optional[str] = None
    groups: Optional[List] = None
    locations: Optional[List[LocationModel]] = None
    providers: Optional[List[str]] = None
    accountIds: Optional[List[AccountIdModel]] = None
    tags: Optional[List[TagModel]] = None
    action: Optional[ActionModel] = None
    events: Optional[List[EventModel]] = None

    class Config:
        extra = 'ignore' 

    @root_validator(pre=True)
    def validate_settings(cls, values):
        if "providers" in values and "locations" in values:
            providers = values.get("providers")
            locations = values.get("locations")
            for location in locations:
                if location.provider not in providers:
                    raise ValueError(
                        f"Location provider '{location.provider}' not in allowed providers list"
                    )
        return values

class FieldModel(BaseModel):
    summary: str
    settings: SettingsModel
    canInsertFieldAfter: bool
    canDelete: bool
    type: str
    title: str
    canConfigure: bool

    class Config:
        extra = "forbid"

class PolicyDocumentModel(BaseModel):
    name: str
    summary: str
    fields: List[FieldModel]
    status: str

    class Config:
        extra = "forbid"

    #@root_validator(pre=True)
    @validator('fields')
    def validate_fields_order(cls, values):
        expected_types = ["if", "with", "then"]
        actual_types = [field.type for field in values]
        if actual_types != expected_types:
            raise ValueError(
                f"Fields must be in order: {', '.join(expected_types)}"
            )
        return values

    @validator("name")
    def validate_name(cls, value):
        if not value:
            raise ValueError("This field is required.")
        return value

class RootModel(BaseModel):
    policy_document: PolicyDocumentModel

    class Config:
        extra = "forbid"

def pydantic_validator_payload(payload):
    """
    # This will raise ValidationError
    model = RootModel(**payload)  # equivalent to unknown=RAISE
    result = model.dict()

    # This will ignore unknown fields
    class RootModel(BaseModel):
        class Config:
            extra = 'ignore'  # equivalent to unknown=EXCLUDE
    result = model.dict()
    """
    try:
        # Using extra = 'forbid' (equivalent to unknown=RAISE)
        validated_data = RootModel(**payload)
        print("--validated_data--", validated_data)
        return {"statusCode": 200, "body": {"message": "validated successfully."}}
    except ValidationError as err:
        # Return the error in the same format as Marshmallow
        return {"statusCode": 400, "body": str(err)}
    except Exception as err:
        return {"statusCode": 400, "body": str(err)}
`

``

``

**Comment by Viicos** [comment]
Could you please format your message properly?
